src/lib.rs

```diff
@@ -198,6 +199,10 @@ create_exception!(retworkx, NoSuitableNeighbors, PyException);
 create_exception!(retworkx, NullGraph, PyException);
 // No path was found between the specified nodes.
 create_exception!(retworkx, NoPathFound, PyException);
+// Prune part of the search tree while traversing a graph.
+import_exception!(retworkx.visit, PruneSearch);
+// Stop graph traversal.
+import_exception!(retworkx.visit, StopSearch);
```

Comment:
Is there a difference between doing this in Python vs Rust? (TBH I didn't even know about this macro).

Reply:
No big difference, initially I did this in Rust but later thought it would be nicer to have a single Python file with these custom exceptions and the visitor base classes.